scripts/ci/run-integration-tests

```diff
@@ -5,49 +5,183 @@
 # replace it with a test runner in the future, please see it more as a proof of
 # concept
 
-set -e
-shopt -s globstar # to allow ** in globs
+## Configuration
+#####################################################################
+
+# Search string passed to `find`
+# Values: String
+SEARCH=${1:-'*'}
+
+# Maximum number of retries
+# Values: n ∈ ℕ
+RETRIES=${RETRIES:-3}
+
+# Stop if file failed $RETRIES times? (turn this off, if you want
+# to test all tests locally but something keeps failing)
+# Values: 0, 1
+STOP_ON_FAIL=${STOP_ON_FAIL:-1}
+
+# Suppresses Cypress output on failures
+# Values: 0, 1
+SUPPRESS_OUTPUT=${SUPPRESS_OUTPUT:-0}
+
+PROXY_PORT=${PROXY_PORT:-4200}
+PROXY_PID=0
 
 PROJECT_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../../" && pwd )"
-RETRIES=3
+CYPRESS_FOLDER="cypress"
+TESTS_FOLDER="tests"
+
+PLUGINS_FOLDER="external_plugins"
+PLUGINS_PATH=${PLUGINS_PATH:-''}
+
+
 
+## Functions
+#####################################################################
+
+# executes Cypress
+#
+# from callee:
+# FILE: file to test
+# OUTPUT_PATH: Directory where results are saved (e.g. "some/path/to/Testfile-cy.js-1")
 function executeCypress {
-  echo "=====> Executing cypress for $1"
-  local testResultPath
-  testResultPath="cypress/$(basename "$1")-result.xml"
-  npm run cypress -- -s "$1" --reporter junit --reporter-options "mochaFile=$testResultPath"
+  npm run --silent cypress -- \
+    --spec "${FILE}" \
+    --config "screenshotsFolder=${OUTPUT_PATH}/screenshots,videosFolder=${OUTPUT_PATH}/videos" \
+    --reporter junit \
+    --reporter-options "mochaFile=${OUTPUT_PATH}/result.xml"
 }
 
+# runs tests for matched file
+#
+# From callee:
+# RELATIVE_PATH: stripped down output from find command (e.g. "pages/Testfile-cy.js")
 function runTestFile {
-  echo "==> Running tests for $1"
-  i="0"
-  while [ $i -lt $RETRIES ]; do
-    i=$((i+1))
-    echo "===> Executing run $i"
-    local exit_code
-
-    # We don't want to let an error in the block below to end the entire script
-    set +e # undo instant error flag
+  echo ""
+  echo "==> Running Tests for ./${TESTS_FOLDER}/$RELATIVE_PATH"
+  local RUN=0
+  local EXIT_CODE=1
+  while [ $EXIT_CODE -ne 0 ] && [ $RUN -lt $RETRIES ]; do
+    RUN=$((RUN+1))
+    local OUTPUT_PATH="${PROJECT_ROOT}/${CYPRESS_FOLDER}/${RELATIVE_PATH}-${RUN}"
 
-    executeCypress "$1"
-    exit_code=$?
+    echo "===> Executing Cypress (${RUN}/${RETRIES})"
+    CYOUT="$(FILE="${PROJECT_ROOT}/tests/${RELATIVE_PATH}" RUN=${RUN} OUTPUT_PATH=${OUTPUT_PATH} executeCypress)"
```

Comment:
I am not sure if I like this, because even If I would not supress output I would not know about the last output when my test hangs for some reason. I would rather drop the feature of silencing then having this drawback, but this is only my point of view

Reply:
sadly it doesnt output anything helpful to stdout :(